plugins/inputs/windows_event_log/wineventlog/wineventlog.go

```diff
@@ -258,19 +295,48 @@ func (w *windowsEventLog) ResubscribeCh() chan struct{} {
 	return w.resubscribeCh
 }
 
+func (w *windowsEventLog) readGaps() []*windowsEventLogRecord {
+	var records []*windowsEventLogRecord
+	for _, r := range w.gapsToRead {
+		if r.IsEndOffsetUnbounded() {
+			continue
+		}
+
+		handle, err := w.openAtRange(r)
+		defer func() {
+			winEventAPI.EvtClose(handle)
+		}()
+		if err != nil {
+			continue
+		}
+		readRecords := w.readFromHandle(handle)
```

Comment:
nit: Right now, the defer call will happen at the end of `readGaps`. Consider moving this into a separate function so the close happens after each read. 

```go
func (w *windowsEventLog) readGap(r state.Range) []*windowsEventLogRecord
```